core/keyboard_nav/cursor_svg.js

```diff
@@ -142,9 +156,19 @@ Blockly.CursorSvg.prototype.setParent_ = function(newParent) {
     return;
   }
   var svgRoot = this.getSvgRoot();
+  var cursorNode = null;
 
   if (newParent) {
-    newParent.appendChild(svgRoot);
+    if (this.isMarker_) {
+      for (var i = 0, childNode; childNode = newParent.childNodes[i]; i++) {
+        if (childNode.className.baseVal == Blockly.CursorSvg.CURSOR_CLASS) {
+          cursorNode = childNode;
+        }
+      }
+      newParent.insertBefore(svgRoot, cursorNode);
+    } else {
+      newParent.appendChild(svgRoot);
```

Comment:
Just to clarify, is this just trying to make sure that the marker cursor comes before the cursor cursor?

Reply:
Yup, otherwise the marker will cover the cursor and you won't be able to see it.